### PR TITLE
- adds marker interfaces for composed types

### PR DESCRIPTION
**Pull request by baywet**
related https://github.com/microsoft/kiota/issues/1411

### CODE REVIEW

File: src/Microsoft.Kiota.Abstractions.csproj [hunk 1]
```diff
@@ -14,7 +14,7 @@
     <EmbedUntrackedSources>true</EmbedUntrackedSources>
     <Deterministic>true</Deterministic>
     <VersionPrefix>1.0.0</VersionPrefix>
-    <VersionSuffix>preview.9</VersionSuffix>
+    <VersionSuffix>preview.10</VersionSuffix>
```
Comment (andrueastman) [inline src/Microsoft.Kiota.Abstractions.csproj:17]: Should be 11

**Review by andrueastman — COMMENTED** [review]
Just need to resolve merge conflicts/rebase

**Comment by baywet** [comment]
@andrueastman solved conflicts

**Review by andrueastman — APPROVED** [review]
👍🏼

**Pull request merged**